### PR TITLE
Built and implemented REST base + major asynchronicity performance optimization

### PR DESCRIPTION
**Pull request by MandarinPine18**
The entirety of this branch was dedicated to getting us out of using WordpressPCL, which doesn't adapt well to the extra flavors of the SNO server. To that end, I've built up a basis for our models to be interconnected with REST API calls so we can communicate with the server directly. Bear in mind that there are **no** major UI changes - this branch is almost purely backend.

To go over changes:
- The folder `Json` contains deserializer classes for a few of the things we're going to be making requests for. They were built simply by transcribing some JSON responses, so I don't know if they are exhaustive, but I think the ones I've written should be. We'll need to repeat the process with every new HTTP endpoint we want to make requests to.
- `RestBase` in the `Models` folder is a base class (C# speak for a superclass) with some convenience items for implementing any REST-based models. It should simplify the process _a lot_ for staff profiles, categories, etc. in the future. The `Article` and `Media` models both inherit from it and implement it fully, as will most future REST-based classes I would think.
- The `Article` and `Media` models both include the `RefreshAsync()` method, implemented from their base class. This is a huge change, as it introduces two major improvements: (1) direct HTTP requests to the server and (2) asynchronicity (!!! :D). That second one is a HUGE performance optimization from before which allows for the app to multitask efficiently. I highly recommend you run release vs. this branch to see the difference.
- There's a few minor tweaks in `ArticleView.xaml`. The gist is that I added fallbacks and null targets to make the app more resilient to (1) binding paths not resolving and (2) bindings returning `null`.
- In `ArticleView.xaml.cs`, there is only one major change - usage of the Dispatcher, which is wildly important for letting this thing run with asynchronicity. We will probably be using the Dispatcher quite a bit in the future.
- I defined the website of our newspaper in `App.xaml.cs`. Going forward, we should **never** be hardcoding that URL anywhere else, instead using the string formatting methods like I do in `RestBase`. The reason is that this app is meant to be a "Stencil" and thus easily adaptable to other base URLs.

I think that's everything. There's quite a bit, but I added full XML documentation for most of the major changes, so you should just be able to hover your mouse over some method if you don't understand what it's doing. Please let me know if anything is unclear at all so I can explain it and add comments for it. Also, don't hesitate to critique me - I really want some criticism here to improve this code more. Thanks :D

### CODE REVIEW

File: BITS-App/Models/RestBase.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        /// </summary>
+        /// <returns>A URI for the server counterpart of the model.</returns>
+        protected Uri GetUri() => new Uri("https://" + REST_URL + endpoint);
+
+        /// <summary>
+        /// Generates a URI dynamically based on the endpoint.
+        /// </summary>
+        /// <param name="args">Formatting arguments if placeholders were in endpoint</param>
+        /// <returns>A URI for the server counterpart of the model.</returns>
+        protected Uri GetUri(object?[] args) => new Uri("https://" + REST_URL + String.Format(endpoint, args));
```
Comment (grkalinich) [inline BITS-App/Models/RestBase.cs]: Possibly could combine these instead of having a whole separate method. Either way words though.

**Review by grkalinich — APPROVED** [review]
Looks great. Hopefully you sleep at some point 💀

**Comment by MandarinPine18** [comment]
What's that word you used? "Sleep". Am I pronouncing it right?

File: BITS-App/Models/RestBase.cs (hunk 1 above)
Comment (MandarinPine18) [reply BITS-App/Models/RestBase.cs]: I think you're right about that. I just looked it up and apparently there's a keyword for parameters called `params` which does just that. Thanks!!

Comment (MandarinPine18) [reply BITS-App/Models/RestBase.cs]: Also, this code-snippet comment thing is cool as heck. I'll be sure to use it later on. Thanks!

**Pull request merged**